packages/user-ui/src/service/user_service.rs

```diff
@@ -0,0 +1,267 @@
+#![allow(non_snake_case)]
+use crate::config;
+use chrono::{DateTime, Utc};
+use dioxus::prelude::*;
+use dioxus_logger::tracing;
+use dioxus_oauth::prelude::FirebaseService;
+use google_wallet::{drive_api::DriveApi, WalletEvent};
+use models::{ApiError, Error, User, UserClient};
+pub enum UserEvent {
+    Signup(String, String, String),
+    Login,
+    Logout,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct UserService {
+    pub cli: Signal<UserClient>,
+    pub firebase: Signal<FirebaseService>,
+    pub firebase_wallet: Signal<google_wallet::FirebaseWallet>,
+
+    pub auth_token: Signal<String>,
+    pub user_id: Signal<i64>,
+    pub email: Signal<String>,
+    pub nickname: Signal<String>,
+}
+
+impl UserService {
+    pub fn init() {
+        let conf = config::get();
+
+        let firebase = FirebaseService::new(
+            conf.firebase.api_key.clone(),
+            conf.firebase.auth_domain.clone(),
+            conf.firebase.project_id.clone(),
+            conf.firebase.storage_bucket.clone(),
+            conf.firebase.messaging_sender_id.clone(),
+            conf.firebase.app_id.clone(),
+            conf.firebase.measurement_id.clone(),
+        );
+
+        #[cfg(feature = "web")]
+        let firebase_wallet = google_wallet::FirebaseWallet::new(
+            conf.firebase.api_key.clone(),
+            conf.firebase.auth_domain.clone(),
+            conf.firebase.project_id.clone(),
+            conf.firebase.storage_bucket.clone(),
+            conf.firebase.messaging_sender_id.clone(),
+            conf.firebase.app_id.clone(),
+            conf.firebase.measurement_id.clone(),
+        );
+
+        #[cfg(not(feature = "web"))]
+        let firebase_wallet = google_wallet::FirebaseWallet::default();
+        let cli = User::get_client(&conf.api_url);
+
+        let user = Self {
+            cli: Signal::new(cli),
+            firebase: use_signal(|| firebase),
+            firebase_wallet: use_signal(|| firebase_wallet),
+
+            auth_token: use_signal(|| "".to_string()),
+            user_id: use_signal(|| 0),
+            email: use_signal(|| "".to_string()),
+            nickname: use_signal(|| "".to_string()),
+        };
+
+        use_context_provider(move || firebase);
+        use_context_provider(move || user);
+    }
+
+    pub async fn google_login(&mut self) -> UserEvent {
```

Comment:
you can simply use GoogleWallet instead of writing all codes.

The reason why incheon-heroes implement itself is that it needed backward-compatibility.

But you can refer to ratel only using GoogleWallet instead of implementing itself.